### ISSUE TITLE
--with-lto still implied by --enable-optimizations in Python 2.7

### ISSUE THREAD

**Issue by d8355cc7-092e-4785-96ee-e450e15672d3** [python/cpython#73952]
BPO | [29766](https://bugs.python.org/issue29766)
--- | :---
Nosy | @gpshead, @hannosch
PRs | <li>python/cpython#1858</li><li>python/cpython#2705</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/gpshead'
closed_at = <Date 2017-09-07.18:44:20.515>
created_at = <Date 2017-03-09.01:15:03.808>
labels = ['build']
title = '--with-lto still implied by --enable-optimizations in Python 2.7'
updated_at = <Date 2017-09-07.18:44:20.514>
user = 'https://github.com/hannosch'
```

bugs.python.org fields:
```python
activity = <Date 2017-09-07.18:44:20.514>
actor = 'gregory.p.smith'
assignee = 'gregory.p.smith'
closed = True
closed_date = <Date 2017-09-07.18:44:20.515>
closer = 'gregory.p.smith'
components = ['Build']
creation = <Date 2017-03-09.01:15:03.808>
creator = 'Hanno Schlichting'
dependencies = []
files = []
hgrepos = []
issue_num = 29766
keywords = []
message_count = 5.0
messages = ['289263', '294730', '294731', '297996', '301608']
nosy_count = 3.0
nosy_names = ['gregory.p.smith', 'Arfrever', 'Hanno Schlichting']
pr_nums = ['1858', '2705']
priority = 'normal'
resolution = 'fixed'
stage = 'commit review'
status = 'closed'
superseder = None
type = 'compile error'
url = 'https://bugs.python.org/issue29766'
versions = ['Python 2.7']
```

</p></details>

**Comment by d8355cc7-092e-4785-96ee-e450e15672d3** [comment]
I think the fix for bpo-28032 wasn't applied correctly to the 2.7 branch.

Compare the change in Python 2.7:

https://github.com/python/cpython/commit/9cbfa79111e7152231556a21af90a220b72ed086#diff-e2d5a00791bce9a01f99bc6fd613a39dL6425

vs. for example Python 3.5:

https://github.com/python/cpython/commit/14c7f71150c94ca35ca913b15c3d0cd236691ed6#diff-e2d5a00791bce9a01f99bc6fd613a39dL6567

In Python 3.5 the Py_LTO='true' line was before the Darwin block and got removed. In Python 2.7 the line was after the block and was left in place.

I'm guessing this was a simply mistake, while backporting the change.

**Comment by gpshead** [comment]
New changeset 1f29cefc87c4c2ee629367ebe97a287d8e0b3e29 by Gregory P. Smith (Hanno Schlichting) in branch '2.7':
bpo-29766: Do not force --with-lto to true for --enable-optimizations (bpo-1858)
https://github.com/python/cpython/commit/1f29cefc87c4c2ee629367ebe97a287d8e0b3e29

**Comment by gpshead** [comment]
indeed, my mistake in the original backport.  thanks!

**Comment by 80036ac5-bb84-4d39-8416-02cd8e51707d** [comment]
>     bpo-29766: Do not force --with-lto to true for --enable-optimizations (bpo-1858)
> 
>     This fixes a faulty backport to the Python 2.7 branch only of http://bugs.python.org/issue28032. Details in the bpo-29766.


This commit is still faulty :) .
It changes configure without configure.ac, so this change is lost for these users who locally adjust configure.ac for some reason and regenerate configure with autoconf.

**Comment by gpshead** [comment]
New changeset abea972d2881a1a04ec265f993bb9b56bbc7b224 by Gregory P. Smith (Hanno Schlichting) in branch '2.7':
[2.7] bpo-29766: Fix configure/.ac to match LTO/enable-optimizations behavior. (bpo-2705)
https://github.com/python/cpython/commit/abea972d2881a1a04ec265f993bb9b56bbc7b224
